### PR TITLE
Upgrade libp2p

### PR DESCRIPTION
**Pull request by willscott**
Upgrade go v0.22.0 libp2p

### CODE REVIEW

**Review by masih — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -5,51 +5,42 @@ go 1.18
 require (
 	github.com/filecoin-project/go-data-transfer v1.15.2
 	github.com/hashicorp/go-multierror v1.1.1
-	github.com/ipfs/go-cid v0.2.0
-	github.com/ipfs/go-datastore v0.5.1
-	github.com/ipfs/go-graphsync v0.13.2
+	github.com/ipfs/go-cid v0.3.2
+	github.com/ipfs/go-datastore v0.6.0
+	github.com/ipfs/go-graphsync v0.13.1
```
Comment (gammazero) [inline go.mod:10]: Is there a need to revert to a previous version of go-graphsync?

File: go.mod [hunk 2]
```diff
@@ -5,51 +5,42 @@ go 1.18
...
-	github.com/libp2p/go-libp2p v0.21.0
-	github.com/libp2p/go-libp2p-core v0.19.1
-	github.com/libp2p/go-libp2p-gostream v0.3.1
-	github.com/libp2p/go-libp2p-peerstore v0.7.1
-	github.com/libp2p/go-libp2p-pubsub v0.7.0
-	github.com/multiformats/go-multiaddr v0.6.0
-	github.com/multiformats/go-multicodec v0.5.0
-	github.com/multiformats/go-multihash v0.2.0
+	github.com/ipld/go-ipld-prime v0.18.0
+	github.com/libp2p/go-libp2p v0.22.0
```
Comment (gammazero) [inline go.mod:13]: At this point, I think we want `github.com/libp2p/go-libp2p v0.23.2` or later.

**Review by gammazero — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (willscott) [reply go.mod:10]: this was the same as what boost / other rebasing against the newer libp2p were using and i figured i'd match that

**Review by gammazero — APPROVED** [review]
Rebase and it's good.

**Comment by gammazero** [comment]
Related:
- https://github.com/filecoin-project/storetheindex/pull/893
- https://github.com/filecoin-project/go-indexer-core/pull/96

**Pull request merged**